firestore/bulkwriter_test.go

```diff
@@ -0,0 +1,233 @@
+package firestore
+
+import (
+	"context"
+	"testing"
+
+	pb "google.golang.org/genproto/googleapis/firestore/v1"
+	"google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc/codes"
+)
+
+type bulkwriterTestCase struct {
+	name string
+	test func(*BulkWriter) (*BulkWriterJob, error)
+}
+
+func TestBulkWriter(t *testing.T) {
+	c, srv, cleanup := newMock(t)
+	defer cleanup()
+
+	docPrefix := c.Collection("C").Path + "/"
+
+	// Create
+	srv.addRPC(
```

Comment:
nit: pulling all this into a setup function that describes what it's doing might help make this test more readable.

Reply:
Done.